core/common/src/Utf8.kt

```diff
@@ -130,6 +130,9 @@ internal fun String.utf8Size(startIndex: Int = 0, endIndex: Int = length): Long
  * Without such a conversion, data written to a [Sink] can not be converted back
  * to a string from which a surrogate pair was retrieved.
  *
+ * More specifically, all code points mapping to UTF-16 surrogates (`U+d800`..`U+dfff`)
+ * will be written as `?` characters (`U+0063`).
```

Comment:
Is this JVM-only behavior or common one?

Reply:
Currently, it's a common one

Reply:
This makes it inconsistent with `String.encodeToByteArray` where the replacement is unspecified, but different.

Reply:
`encodeToByteArray` call on a string with a low surrogate only results in a byte array containing `?`: https://pl.kotl.in/uFX3YULin